src/gridstack.js

```diff
@@ -1435,8 +1435,13 @@
     }
   };
 
-  GridStack.prototype.addWidget = function(el, x, y, width, height, autoPosition, minWidth, maxWidth,
-    minHeight, maxHeight, id) {
+  GridStack.prototype.addWidget = function(el, x, y, width, height, autoPosition, minWidth, maxWidth, minHeight, maxHeight, id) {
+
+    // instead of passing all the params, the user might pass an object with all fields instead, if so extract them and call us back
+    if (typeof x === 'object') {
```

Comment:
this will fail if we pass `null` as a value of `x`, as `typeof null === 'object'` returns true.